sdk/test-utils/recorder/src/baseRecorder.ts

```diff
@@ -74,11 +63,22 @@ export abstract class BaseRecorder {
    * @memberof BaseRecorder
    */
   protected filterSecrets(content: any): any {
+    let updatedContent = content;
+    if (typeof content !== "string") {
```

Comment:
Does this check mean that we won't ever have default customization for string content?

Reply:
`filterSecretsFromStrings` calls the customizations on the string content already.

For JSON, customizations are not being called on the overall recording, `filterSecretsRecursivelyFromJSON` parses recursively before calling the customizations.